docs/dev/stream/state/broadcast_state.md

```diff
@@ -0,0 +1,279 @@
+
+## Provided APIs
+
+To show the provided APIs, we will start with an example before presenting their full functionality. As our running 
+example, we will use the case where we have a stream of objects of different colors and shapes and we want to find pairs
+of objects of the same color that follow a certain pattern, *e.g.* a rectangle followed by a triangle. We assume that
+the set of interesting patterns evolves over time. 
+
+In this example, the first stream will contain elements of type `Item` with a `Color` and a `Shape` property. The other
+stream will contain the `Rules`.
+
+Starting from the stream of `Items`, we just need to *key it* by `Color`, as we want pairs of the same color. This will
+make sure that elements of the same color end up on the same physical machine.
+
+{% highlight java %}
+// key the shapes by color
+KeyedStream<Item, Color> colorPartitionedStream = shapeStream
+                        .keyBy(new KeySelector<Shape, Color>(){...});
+{% endhighlight %}
+
+Moving on to the `Rules`, the stream containing them should be broadcasted to all downstream tasks, and these tasks 
+should store them locally so that they can evaluate them against all incoming `Items`. The snippet below will i) broadcast 
+the stream of rules and ii) using the provided `MapStateDescriptor`, it will create the broadcast state where the rules
+will be stored.
+
+{% highlight java %}
+
+// a map descriptor to store the name of the rule (string) and the rule itself.
+MapStateDescriptor<String, Rule> ruleStateDescriptor = new MapStateDescriptor<>(
+				"RulesBroadcastState",
+				BasicTypeInfo.STRING_TYPE_INFO,
+				TypeInformation.of(new TypeHint<Rule>() {})
+		);
+		
+// broadcast the rules and create the broadcast state
+BroadcastStream<Rule> ruleBroadcastStream = ruleStream
+                        .broadcast(ruleStateDescriptor);
+{% endhighlight %}
+
+Finally, in order to evaluate the `Rules` against the incoming elements from the `Item` stream, we need to:
+    1) connect the two streams and 
+    2) specify our match detecting logic. 
+
+Connecting a stream (keyed or non-keyed) with a `BroadcastStream` can be done by calling `connect()` on the 
+non-broadcasted stream, with the `BroadcastStream` as an argument. This will return a `BroadcastConnectedStream`, on 
+which we can call `process()` with a special type of `CoProcessFunction`. The function will contain our matching logic. 
+The exact type of the function depends on the type of the non-broadcasted stream: 
+ - if that is **keyed**, then the function is a `KeyedBroadcastProcessFunction`. 
+ - if it is **non-keyed**, the function is a `BroadcastProcessFunction`. 
+ 
+ Given that our non-broadcasted stream is keyed, the following snippet includes the above calls:
+
+<div class="alert alert-info">
+  <strong>Attention:</strong> The connect should be called on the non-broadcasted stream, with the `BroadcastStream`
```

Comment:
The BroadcastStream actually isn't formatted as a code font after rendering.

Reply:
What happens if the user calls connect on the wrong (broadcasted) stream?
Is there an exception? If so it might make sense to clarify that here.

Reply:
It is not even available as an option to call `connect()` in this case. The `broadcast(descriptor)` will give you a `BroadcastStream` back that has no available transformations. You can only use it as an argument for a `connect()`.